### PR TITLE
BUG: Fixes KeyError when indexes don't overlap.

### LINKED ISSUE #10291: KeyError: '__dummy__'  for pd.crosstab in pandas
get ~~ KeyError: '__dummy__' ~~ when I run the following:

    np.random.seed(seed = 99)
    s = np.random.randint(1,10,200)
    s = pd.Series(np.where(s > 9, np.nan, s))
    s1 = s[:100]
    s2 = s[100:]
    pd.crosstab(s1, s2)

    KeyError: '__dummy__

### PR DESCRIPTION
**Pull request by tshauck**
Closes #10291

### CODE REVIEW

**Comment by tshauck** [comment]
#10291 & #12133 for easy linking.

**Comment by TomAugspurger** [comment]
Can you edit your first post to include "Closes #10291, Closes #12133" so that Github will auto-close those issues when we merge?

Also would need a release note in `doc/source/whatsnew/v0.18.0` under bug fixes and maybe add a note to the docstring of `crosstab` explaining that this is the outcome for non-overlapping indices. Thanks!

File: pandas/tools/tests/test_pivot.py [hunk 1]
```diff
@@ -892,6 +892,15 @@ def test_categorical_margins(self):
...
         tm.assert_frame_equal(table, expected)
 
+    def test_crosstab_no_overlap(self):
+        # GS 10291
+
+        s1 = pd.Series([1, 2, 3], index=[1, 2, 3])
+        s2 = pd.Series([4, 5, 6], index=[4, 5, 6])
+
+        actual = crosstab(s1, s2)
+        self.assertTrue(actual.empty)
```
Comment (jreback) [inline pandas/tools/tests/test_pivot.py]: construct the actual result and use ``assert_frame_equal``

**Comment by jreback** [comment]
you need a test to cover #12133

**Comment by jreback** [comment]
pls add a whatsnew as well

**Comment by tshauck** [comment]
Thought it may be easier in the near term to reduce the scope of this change to fixing the issue associated with #10291 and not attempt a fix at #12133.

I've added #12133 to my todo list so I'll try to get back to it.

**Comment by jreback** [comment]
thanks @tshauck

**Comment by Winand** [comment]
`if values_passed and not values_multi and not table.empty:`
Excuse me, but how does this cover #12133? index names are dropped earlier in `if not dropna:` block. We need to pass names to `from_arrays`:

```
m = MultiIndex.from_arrays(cartesian_product(table.index.levels), names=table.index.names)
...
m = MultiIndex.from_arrays(cartesian_product(table.columns.levels), names=table.columns.names)
```

**Comment by jreback** [comment]
if u read the comments about it does not cover 12133

**Pull request closed without merging**